json/src/de.rs

```diff
@@ -885,7 +927,7 @@ pub struct StreamDeserializer<T, Iter>
     where Iter: Iterator<Item=io::Result<u8>>,
           T: de::Deserialize
 {
-    deser: Deserializer<Iter>,
+    deser: DeserializerImpl<read::IteratorRead<Iter>>,
```

Comment:
`StreamDeserializer` is not specialized. Probably if you are using `StreamDeserializer` you have a stream of things and not a `&str` anyway.